### ISSUE TITLE
Automatic size of grid for `resume-skill-item`

### ISSUE THREAD

**Issue by vslavkin** [DeveloperPaul123/modern-cv#84]
I wanted to write "programming languages" as a skill item, but it doesn't fit, so it does a not-so-pleasant line-break. 
While an argument with the column lengths could be passed, I wonder if there's a way to apply the max auto size of the column to the multiple grid one may have in a document.
If that couldn't be done, then the function could instead receive a list of categories and a list of lists of items, and put everything in one big grid.

**Comment by vslavkin** [comment]
Now that I think of it, the items don't have to be in a list. They could be just content (Except if you wanted to parse them for something), and it would give more freedom to the user

**Comment by DeveloperPaul123** [comment]
Thanks for filing this. If you have any screenshots or images of what exactly you're envisioning, that would be very helpful.

**Comment by vslavkin** [comment]
Sorry, forgot about this, but made a small demo:
This would be the desired result ![image](https://github.com/user-attachments/assets/d6542e62-bad1-4937-aa11-a8e26c8ae0f1)
And this is what is currently happening: ![image](https://github.com/user-attachments/assets/207f786c-91e7-4665-86d8-fd3539616537)

I understand that some people will prefer the former and others the latter, but it would be great to offer an option in `#resume-skill-item` to toggle the feature

The code used for the former is this:
```typst
#grid(
    columns: (auto, auto),
    gutter: 10pt,
    [== Programming Languages],
    [*Python*, *C*, C++, Java, Lisp, Html, CSS, JavaScript, Rust, SQL, Pascal, Lua, Bash,],
    [== Spoken languages],
    [*Spanish (Native)*, *English (B2)*],
    [== Programs],
    [*Emacs*, *LaTeX*, git, Godot, Docker, Grafana, Unity, KiCad, Blender, DaVinci Resolve, Inkscape, OnShape])
```

While it works here, I don't know how to implement this, maintaining the width of the first column the same across all function calls.
![image](https://github.com/user-attachments/assets/ac5fdd78-49ee-445e-a2a6-6c2773753825)

![image](https://github.com/user-attachments/assets/c1128584-6514-4910-b322-01dccbfc58d5)

The easy way is to add an optional argument to the `#resume-skill-item` function for setting the width of the first column, but I'd say that the best way is using the auto setting. 
If it was possible to save the width of the auto width to a variable, I think it may be doable, but I don't know enough typst yet.

**Comment by xd003** [comment]
This is must have imo, i was just about to open a github issue for this. I totally agree with the desired result mentioned above. Moreover, if i happen to remove the Spoken languages line from the default template, it looks totally weird

![Image](https://github.com/user-attachments/assets/bc49e01d-d2da-4b8a-b70a-cfbaaa42065e)
Would really appreciate to have this issue fixed

**Comment by vslavkin** [comment]
Yeah, it's indeed tricky. What you've got there looks good, is there a branch to test it out?
The only other ways I can think of would be, doing it in a single function/grid, or measuring the size of the text every time the function is called and render it after all the function calls have been made, but I don't know if that's possible in typst

**Comment by DeveloperPaul123** [comment]
I was able to make a new function that lays everything out in a grid. Here's the result:

![Image](https://github.com/user-attachments/assets/cb1b75b7-a43d-447d-b35f-e163b787dec5)

And the typst to generate it:
```typ
#resume-skill-grid(
  categories_with_values: (
    "Programming Languages": (
      strong("C++"),
      strong("Python"),
      "Rust",
      "Java",
      "C#",
      "JavaScript",
      "TypeScript",
    ),
    "Spoken Languages": (
      strong("English"),
      "Spanish",
      "Greek",
    ),
    "Programs": (
      strong("Excel"),
      "Word",
      "Powerpoint",
      "Visual Studio", 
      "git",
      "Zed"
    ),
    "Really Really Long Long Long Category": (
      "Thing 1",
      "Thing 2",
      "Thing 3"
    )
  ),
)
```
